update-versions.py

```diff
@@ -103,16 +104,27 @@ def nix_prefetch(args):
 g = github.Github(auth=auth)
 repo = g.get_repo("hashicorp/terraform")
 # TODO: Drop "v" prefix first
-# TODO: Combine the output of the following endpoints to get a list of active
-# versions
-# https://api.github.com/repos/hashicorp/terraform/tags
-# https://endoflife.date/api/terraform.json
-releases = list(filter(is_stable, repo.get_releases()))
+
+# Craft list of maintaned releases by filtering the release list with the EOL data
+eol_data = requests.get("https://endoflife.date/api/terraform.json").json()
+maintained_cycles = [cycle['cycle'] for cycle in eol_data if not cycle['eol']]
+maintained_releases = list(filter(lambda version: is_stable(version, maintained_cycles), repo.get_releases()))
+
+# Read the current versions and drop unmaintaned ones
 current_versions = read_versions()
+maintained_versions = set(release.title.lstrip('v') for release in maintained_releases)
+filtered_versions = {}
+for k, v in current_versions.items():
+    if k in maintained_versions:
+        filtered_versions[k] = v
+    else:
+        print(f"Dropping support for {k}, it is no longer maintaned")
```

Comment:
@oscar-izval while I like the idea of having this script handle both new and deprecated versions, I believe it would be best to limit the scope of this PR to only new versions. The main reason I'm proposing this is that removing some entries from the `versions.json` file has some implications for the release process; for example, if these scripts only add new versions, that could be considered a minor release, but if both new versions are added and old versions are removed from the `versions.json` file, that should be considered a major version; unfortunately, I don't think we have enough logic in the `update` workflow to handle all of these edge cases. What do you think?

Reply:
That's absolutely right, let's tackle version deprecation in a separate PR. Thanks for the suggestion!